avc/src/finish_detector/finish_detector.cpp

```diff
@@ -39,31 +44,31 @@ void finish_detector::ImageCB(const sensor_msgs::ImageConstPtr& msg) {
 	}
 	
 	frame = cv_ptr->image;
-
-    for(int r = 0; r < frame.rows; r++) {
-        unsigned char* row = frame.ptr<unsigned char>(r);
-        for(int c = 0; c < frame.cols * frame.channels(); c+= frame.channels()) {
-            auto& blue = row[c];
-            auto& green = row[c+1];
-            auto& red = row[c+2];
-            if(blue < 50 && green < 50 && red > 200) {
-                blue = green = red = 255;
-            } else {
-                blue = green = red = 0;
-            }
-        }
-    }
-    cvtColor(frame, frame, CV_BGR2GRAY);
-
-    auto count = countNonZero(frame);
+	Mat frameHSV;
+	cvtColor(frame, frameHSV, CV_BGR2HSV);
+        Mat blurredImage;
+        GaussianBlur(frameHSV, blurredImage, Size{7, 7}, 7);
+        Mat redMask1 = Mat::zeros(blurredImage.rows, blurredImage.cols, CV_8U);
+        Mat redMask2 = Mat::zeros(blurredImage.rows, blurredImage.cols, CV_8U);
+        inRange(blurredImage, red_low1, red_high1, redMask1);
+        inRange(blurredImage, red_low2, red_high2, redMask2);
+	Mat mask;
+	bitwise_or(redMask1, redMask2, mask);
+	imshow("filtered", mask);
+	waitKey(10);
+
+    auto count = countNonZero(mask);
 
     if(state == LOW && count > 1000) {
         state = HIGH;
     } else if(state == HIGH && count < 1000) {
         // We crossed the line!
         state = LOW;
-        number_of_crosses++;
-	ROS_INFO_STREAM("Finish line crossed - " << to_string(number_of_crosses));
+	if (ros::Time::now() > lastCross + ros::Duration(10)) {
+            number_of_crosses++;
+	    lastCross = ros::Time::now();
+	}
+	NODELET_FATAL_STREAM("Finish line crossed - " << to_string(number_of_crosses));
```

Comment:
We should probably figure out why we had to use FATAL log levels at comp. Obviously, this is not the right way to be displaying information.

Reply:
Obviously ROS_WARN_STREAM_DELAYED_THROTTLE_NAMED is the superior macro

Reply:
Not gonna lie, didn't even know that was a valid combination of modifiers